dm/pkg/checker/conn_checker.go

```diff
@@ -46,7 +48,7 @@ func (c *connNumberChecker) check(ctx context.Context, checkerName string) *Resu
 	result := &Result{
 		Name:  checkerName,
 		Desc:  "check if connetion concurrency exceeds database's maximum connection limit",
-		State: StateFailure,
+		State: StateSuccess,
```

Comment:
if it's `StateSuccess` here, then on every `return on error`, we will return success

Reply:
we have `markCheckError(result, err)` before every error return:
```
func markCheckError(result *Result, err error) {
	if err != nil {
		state := StateFailure
		if utils.OriginError(err) == context.Canceled {
			state = StateWarning
		}
		// `StateWarning` can't cover `StateFailure`.
		if result.State != StateFailure {
			result.State = state
		}
		result.Errors = append(result.Errors, &Error{Severity: state, ShortErr: err.Error()})
	}
}
```
Actually, this initial value does not matter because it is always reset before being returned.